### PR TITLE
DAT-3438 case insensitive NS_FILE name in infobox

### PR DESCRIPTION
**Pull request by dianafa**
https://wikia-inc.atlassian.net/browse/DAT-3428

### CODE REVIEW

**Comment by idradm** [comment]
Awesome, :+1: for tests

**Comment by dianafa** [comment]
per offline discussion: @SebastianMarzjan suggested that we should make case insensitive only the first letter as it follows the MW convention of file naming

File: includes/wikia/helpers/tests/FileNamespaceSanitizeHelperTest.php [hunk 1]
```diff
@@ -264,6 +264,27 @@ public function testSanitizeFilenameDataProvider() {
...
+				null,
+				'Luke+1.jpg',
+				'Link to filename with upper and lower case letters in namespace'
+			],
+			[
+				'[[файл:Luke+1.jpg]]',
+				'ru',
+				null,
+				'Luke+1.jpg',
+				'Link to filename with lower case multibyte letter in namespace'
```
Comment (sebastianmarzjan) [inline includes/wikia/helpers/tests/FileNamespaceSanitizeHelperTest.php:279]: :+1:

File: includes/wikia/helpers/FileNamespaceSanitizeHelper.php [hunk 2]
```diff
@@ -39,9 +39,15 @@ private function getFilePrefixRegex( $contLang ) {
...
 			//be able to match user-provided file namespaces that may contain both underscores and spaces
-			$fileNamespaces = array_map(function( $namespace ) {
-				return mb_ereg_replace('_', '(_|\ )', $namespace);
-			}, $fileNamespaces);
+			$fileNamespaces = array_map( function( $namespace ) {
+				return mb_ereg_replace( '_', '(_|\ )', $namespace );
+			}, $fileNamespaces );
+
+			//be able to match both upper- and lowercase first letters of the namespace
+			foreach ( $fileNamespaces as $namespace ) {
```
Comment (idradm) [inline includes/wikia/helpers/FileNamespaceSanitizeHelper.php]: Changing array you are iterating seems unpredictable. Could you prepare it separately and the merge them? Additionally it seems like perfect array_map use case.

**Comment by idradm** [comment]
LGTM

**Comment by nikodamn** [comment]
@tomasznapieralski resolved your comments. Please take a quick look

**Pull request merged**